tests-ui/tests/store/modelStore.test.ts

```diff
@@ -0,0 +1,79 @@
+import { setActivePinia, createPinia } from 'pinia'
+import { useModelStore } from '@/stores/modelStore'
+import { api } from '@/scripts/api'
+
+// Mock the api
+jest.mock('@/scripts/api', () => ({
+  api: {
+    getModels: jest.fn(),
+    viewMetadata: jest.fn()
+  }
+}))
+
+describe('useModelStore', () => {
+  let store: ReturnType<typeof useModelStore>
+
+  beforeEach(() => {
+    setActivePinia(createPinia())
+    store = useModelStore()
+  })
+
+  it('should load models', async () => {
+    ;(api.getModels as jest.Mock).mockResolvedValue([
+      'sdxl.safetensors',
+      'sdv15.safetensors',
+      'noinfo.safetensors'
+    ])
+    ;(api.viewMetadata as jest.Mock).mockImplementation((_, model) => {
+      if (model === 'noinfo.safetensors') {
+        return Promise.resolve({})
+      }
+      return Promise.resolve({
+        'modelspec.title': `Title of ${model}`,
+        display_name: 'Should not show',
+        'modelspec.architecture': 'stable-diffusion-xl-base-v1',
+        'modelspec.author': `Author of ${model}`,
+        'modelspec.description': `Description of ${model}`,
+        'modelspec.resolution': '1024x1024',
+        trigger_phrase: `Trigger phrase of ${model}`,
+        usage_hint: `Usage hint of ${model}`,
+        tags: `tags,for,${model}`
+      })
+    })
```

Comment:
This mock should live outside test body as other tests also depend on this mock.

Reply:
I tried putting it in the initial `jest.mock` call at the top, it was ignored. I tried putting it in the describe block outside the `it` block, it was ignored. I tried putting it in the beforeEach, it was ignored. I don't know how to make it apply from outside of an `it` block. They do however run linearly so being in the first block applies it to all of them.

Reply:
I think you can wrap it as a function and call it in test case that depends on it. Doing it in a test case and let other subsequent test case depends on it should be avoided as this makes us unable to disable individual test.

Reply:
fair enough, done